WIP: Merge mathematical logic and infrastructure improvements

vaibhavmehrotraml opened the pull request:
Primary Contributions:
- Change logic in acoustic_utils to match Ben's visualisations
- Parallelize data download and processing
- Change orca-hls-utils node to @vaibhavmehrotraml's implementation

WIP:
- Update documentation 
- Test edge cases
- Cost analysis

vaibhavmehrotraml committed:
feat(requirements): Update hls-utils to use parallelized version

vaibhavmehrotraml committed:
refactor(acoustic_util.py): PEP formatting

vaibhavmehrotraml committed:
feat(pipeline): Update generate_psds to accomodate multiprocessing

vaibhavmehrotraml committed:
feat(pipeline): Add safe and fast mode functionality

vaibhavmehrotraml committed:
feat(utils): Add simple visualisation pipeline

vaibhavmehrotraml committed:
feat(utils): Add Cost analysis and logging script

vaibhavmehrotraml committed:
fix(utils): Change logging name

vaibhavmehrotraml committed:
refactor(utils): Reformat logging

vaibhavmehrotraml committed:
refactor(utils): Add duration to logging

vaibhavmehrotraml committed:
fix(utils): Move typecaasting one level down

vaibhavmehrotraml committed:
fix(utils): Move typecaasting one level down

vaibhavmehrotraml committed:
fix(utils): Fix format

vaibhavmehrotraml committed:
fix(utils): Fix format

vaibhavmehrotraml committed:
fix(utils): Fix format

vaibhavmehrotraml committed:
debug(cost_analysis): Add shape to debug

zachprice12 committed:
Adding updates to acoustic_util, pipeline, hydrophone

vaibhavmehrotraml committed:
feat(test_3.py): Add time benchmarking

vaibhavmehrotraml committed:
fix(test_3.py): Remove hard-coded paths

vaibhavmehrotraml committed:
debug(pipeline): Add start and end time

zachprice12 committed:
Removing pipeline_3

vaibhavmehrotraml committed:
debug(pipeline): Add start and end time deeper

vaibhavmehrotraml committed:
debug(pipeline): Add debugging counter

vaibhavmehrotraml committed:
debug(pipeline): Add debugging counter

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
debug(pipeline): Fix timezone

vaibhavmehrotraml committed:
fix(cost_analysis): Change to port townsend:

vaibhavmehrotraml committed:
src/orcasound_noise/pipeline/pipeline.py

vaibhavmehrotraml committed:
Prep for prod

vaibhavmehrotraml committed:
requirements: Fix pandas version

vaibhavmehrotraml committed:
Merge branch 'fix/vm_downloads' into vai-zach-merge

vaibhavmehrotraml committed:
feat(visualise): Update viz script to use Zach's test_3

vaibhavmehrotraml committed:
feat(pipeline): Destroy objects to free up memory

vaibhavmehrotraml committed:
fix(pipeline): Remove pqt_folder argument

zprice12 reviewed (approved):
Looks good to me. We could try to implement the fastparquet write method if having RAM problems, but saw you added the delete for the lists of dataframes which should help with RAM

ttan27 committed:
add tutorial

zachprice12 committed:
Moved Tim's tutorial into the README and added a pipeline overview

zachprice12 committed:
Documentation- Added to the README, pipeline README, and in-line commented

zprice12 committed:
Adding readme images to the repo

zprice12 committed:
Added images to the README

zprice12 committed:
Delete pages/doc_images/readme_bb.png

zprice12 committed:
Delete pages/doc_images/readme_spec.png

zachprice12 committed:
Adjusted image size in README

zachprice12 committed:
Added a broadband graphing function

zprice12 committed:
Updated broadband image in the README

ttan06 committed:
add references

valentina-s commented on src/orcasound_noise/pipeline/acoustic_util.py:
```diff
@@ -217,21 +292,22 @@ def ancient_ambient(df):
 
 def spec_plot(df, sr=48000, hop_length=256):
```
There is a function `spec_plot` and `plot_spec`: very confusing.

valentina-s commented on src/orcasound_noise/pipeline/acoustic_util.py:
```diff
@@ -397,35 +474,87 @@ def plot_noise(testdf, name, output_path=None, save = False):
+
+    Returns: Time series of broadband levels.
+    """
+
```
add `plt.xticks(rotation = 45)` so that the xticks are visible

valentina-s reviewed (approved):
I have a few comments inline. Some of them became outdated with the new pushes. You can use the option 'Show outdated'. They are mainly about making the function names in quotes so that they are more readable. And a few small things. Thanks!

vaibhavmehrotraml committed:
chore(): Cleanup and att github/linked

valentina-s committed:
Update README.md

adding github installation instructions, and some quotes to improve reading of functions

valentina-s committed:
spectrograms and graphs -> spectrograms and time series plots

valentina-s committed:
rotating xticks in plot

valentina-s commented:
Thank you @zachprice12 @vaibhavmehrotraml and @ttan27 on your contributions!
